### PR TITLE
feat(its)!: add operator cap to its and allow operator to overwrite flow limits

### PR DESCRIPTION
**Pull request by Foivos**
[AXE-6863]

[AXE-6863]: https://axelarnetwork.atlassian.net/browse/AXE-6863?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/lucky-kangaroos-relax.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@axelar-network/axelar-cgp-sui': minor
+---
+
+Add operator cap to its and allow operator to overwrite flow limits
```
Comment (milapsheth) [inline .changeset/lucky-kangaroos-relax.md:5]:
```suggestion
Add operator cap to ITS and allow operator to overwrite flow limits
```

File: move/its/sources/its.move [hunk 2]
```diff
@@ -255,6 +261,8 @@ public fun burn_as_distributor<T>(
     );
 }
 
+// This is the entrypoint for operators to set the flow limits of their tokens
+// (tokenManager.setFlowLimit on EVM)
```
Comment (milapsheth) [inline move/its/sources/its.move:265]:
```suggestion
/// Set the flow limit for a given tokenId
```
Use `///` for docstrings. Also mention how to remove the flow limit.

File: move/its/sources/its.move [hunk 3]
```diff
@@ -270,6 +278,22 @@ public fun set_flow_limit<T>(
     );
 }
 
+// This is the entrypoint for operators to set the flow limits of their tokens
+// (interchainTokenService.setFlowLimits on EVM)
```
Comment (milapsheth) [inline move/its/sources/its.move:282]: same

File: move/its/sources/its.move [hunk 4]
```diff
@@ -270,6 +278,22 @@ public fun set_flow_limit<T>(
...
+public fun set_flow_limit_as_operator<T>(
```
Comment (milapsheth) [inline move/its/sources/its.move:283]: this should be the default `set_flow_limit` method. The other method can be renamed to `set_flow_limit_by_token_distributor`

File: move/its/sources/its.move [hunk 5]
```diff
@@ -270,6 +278,22 @@ public fun set_flow_limit<T>(
...
+    self: &mut ITS,
+    _: &OperatorCap,
+    token_ids: TokenId,
```
Comment (milapsheth) [inline move/its/sources/its.move:286]:
```suggestion
    token_id: TokenId,
```

File: move/its/sources/types/coin_management.move [hunk 6]
```diff
@@ -129,6 +129,14 @@ public(package) fun set_flow_limit<T>(
     flow_limit: u64,
 ) {
     assert!(self.operator.contains(&channel.to_address()), ENotOperator);
+    self.set_flow_limit_permissionless(flow_limit);
+}
+
+/// Adds a rate limit to the `CoinManagement`.
+public(package) fun set_flow_limit_permissionless<T>(
```
Comment (milapsheth) [inline move/its/sources/types/coin_management.move:136]:
```suggestion
public(package) fun set_flow_limit_internal<T>(
```

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@axelar-network/axelar-cgp-sui",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (milapsheth) [inline package.json:3]: this shouldn't be changed manually

File: move/its/sources/its.move (hunk 2 above)
Comment (Foivos) [reply move/its/sources/its.move:265]: done

File: move/its/sources/its.move (hunk 4 above)
Comment (Foivos) [reply move/its/sources/its.move:283]: use `set_flow_limit_as_token_operator` instead

File: move/its/sources/its.move (hunk 5 above)
Comment (Foivos) [reply move/its/sources/its.move:286]: fixed

File: package.json (hunk 7 above)
Comment (Foivos) [reply package.json:3]: removed

**Review by milapsheth — APPROVED** [review]

**Pull request merged**